packages/openmrs/src/Adaptor.js

```diff
@@ -134,18 +102,29 @@ export function createEncounter(data, callback = false) {
  * @param {function} [callback] - Optional callback to handle the response
  * @returns {Operation}
  */
-export function get(path, query, callback = false) {
-  return state => {
-    const { instanceUrl } = state.configuration;
-    const resolvedPath = expandReferences(path)(state);
-    const resolvedQuery = expandReferences(query)(state);
-    const urlPath = `${instanceUrl}/ws/rest/v1/${resolvedPath}`;
-
-    return agent
-      .get(urlPath)
-      .query(resolvedQuery)
-      .then(response => handleResponse(response, state, callback))
-      .catch(handleError);
+export function get(path, query, callback = s => s) {
+  return async state => {
+    const [resolvedResource, resolvedQuery = {}] = expandReferences(
+      state,
+      path,
+      query
+    );
+
+    console.log(`Preparing get operation...`);
+    const response = await request(
+      state,
+      'GET',
+      `/ws/rest/v1/${resolvedResource}`,
+      {},
+      resolvedQuery
+    );
+    if (response.statusCode === 200) {
+      console.log(`Get operation successful for ${resolvedResource}...`);
+    } else {
+      console.log(`Get operation failed for ${resolvedResource} ...`);
```

Comment:
what happens on error? i'd expect that this function should make the job exit with an error status if it's not a 200.

Reply:
Looking at this now, the request helper should be throwing errors for any 400+ requests, which will bubble up and kill the job. So I think we're looking good right now!